driver/src/main/java/org/neo4j/driver/internal/async/inbound/InboundMessageDispatcher.java

```diff
@@ -140,13 +132,14 @@ public void handleFailureMessage( String code, String message )
         if ( ErrorUtil.isFatal( currentError ) )
         {
             // we should not continue using channel after a fatal error
-            // fire error event back to the pipeline and avoid sending ACK_FAILURE
+            // fire error event back to the pipeline and avoid sending RESET
             channel.pipeline().fireExceptionCaught( currentError );
             return;
         }
 
-        // try to write ACK_FAILURE before notifying the next response handler
-        ackFailureIfNeeded();
+        // write a RESET to "acknowledge" the failure
+        queue( new ResetResponseHandler( this, null ) );
```

Comment:
Maybe better to create a constructor that only takes an `InboundMessageDispatcher` and uses `null` inside `ResetResponseHandler`. Just to not pass `null` explicitly as the second parameter.